src/main/java/org/zephyrsoft/sdb2/gui/MainWindow.java

```diff
@@ -833,52 +938,37 @@ private static void setEnabledIfNotNull(JComponent component, boolean enabled) {
 	protected void handleSongsListSelectionChanged(ListSelectionEvent e) {
 		// only the last event in a row should fire these actions (check valueIsAdjusting)
 		if (!e.getValueIsAdjusting()) {
-			saveSong();
-			songsListSelected = songsList.getSelectedValue();
-			loadSong();
-		}
-	}
-	
-	private void saveSong() {
-		if (songsListSelected != null) {
-			saveSongWithoutChangingGUI();
-			clearSongData();
-			setSongEditingEnabled(false);
-			// disable buttons
-			btnDeleteSong.setEnabled(false);
-			btnSelectSong.setEnabled(false);
-			btnExportPdfSelected.setEnabled(false);
+			Song song = songsList.getSelectedValue();
+			LOG.debug("SongListselectionChange to {}", song);
+			setSelectedSong(song, true);
 		}
 	}
 	
 	private void saveSongWithoutChangingGUI() {
-		if (songsListSelected != null) {
-			saveSongData(songsListSelected);
-		}
-	}
-	
-	private void loadSong() {
-		if (songsListSelected != null) {
-			loadSongData(songsListSelected);
-			setSongEditingEnabled(true);
-			// enable buttons
-			btnDeleteSong.setEnabled(true);
-			btnSelectSong.setEnabled(true);
-			btnExportPdfSelected.setEnabled(true);
+		if (!isSongDataChanged()) {
+			return;
 		}
+		
+		LOG.debug("saveSongData: {}", selectedSong.getTitle());
+		controller.getSongsController().updateSong(songFromGUI());
 	}
 	
 	private void applyFilter() {
-		saveSong();
+		// Filter:
+		final String filterText = textFieldFilter.getText();
```

Comment:
Why is this extracted? It's only used once...

Reply:
Do you mean the final String filterText? I use it later on the AWT event dispatching thread.